src/__test__/components/data-management/DataManagementIntercept.test.jsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import { screen, render } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { useRouter } from 'next/router';
+import fake from '__test__/test-utils/constants';
+import thunk from 'redux-thunk';
+import configureStore from 'redux-mock-store';
+import userEvent from '@testing-library/user-event';
+import DataManagementIntercept from '../../../components/data-management/DataManagementIntercept';
+
+jest.mock('next/router', () => ({
+  __esModule: true,
+  useRouter: jest.fn(),
+}));
+
+const mockRouter = {
+  pathname: '/data-processing',
+  push: jest.fn(),
+};
+useRouter.mockReturnValue(mockRouter);
+
+const mockStore = configureStore([thunk]);
+jest.mock('');
+const store = mockStore({});
+describe('Data Management Intercept', () => {
```

Comment:
Add tests for : 
- Discarding the the changes and continuing to the path
- Dismissing the modal using the close button (the x on the top right)

Reply:
There is already a test - Dispatches routes correctly for continuing the route in AppRouteProvider.test, which is basically testing the onContinue function that is passed to DataManagementIntercept as well. `onDismissIntercept` is also tested, which is used for `onCancel`.

Reply:
Or am i missing something?

Reply:
The one that we want to test here is the interface, the the buttons in `DataManagementIntercept` dispatch the correct actions. You have tested one of them, the re-process button, but there are 2 more actions a user can take, clicking on the discard button and clicking on the close button. So those are the 2 interactions that we want to test for,

Reply:
Oh I see that you have tested for clicking the `continue` in the first case. Can you split the test case to make it clearer?

Reply:
Then there is only 1 test case left to handle, which is clicking the `x` button on the top right part of the modal.

Reply:
Aside from checking `onDismissIntercept` is triggered, is it possible to check if the element does not exist?